acme/client.py

```diff
@@ -0,0 +1,234 @@
+        self.headers = dict()
+
+        # Automatically login.
+        if email is not None and password is not None:
+            self.login(email, password)
+
+
+    def _make_request(self, method: str, endpoint: str, data: dict=None, headers: dict=None) -> dict:
+        """Make the request
+
+        Parameters
+        ----------
+        method : str
+            Method name (GET, POST, PUT, UPDATE, DELETE)
+        endpoint : str
+            URL slug to add to the base url.
+        data : dict, optional
+            Key/value pairs of data to send., by default None
+        headers : dict, optional
+            Headers to add to the request, by default None
+
+        Returns
+        -------
+        dict or request.Response
+            Decoded JSON response, or raw response.
+
+        Raises
+        ------
+        Exception
+            Raised when there is an issue.
+        """
+
+        # Cast to uppercase, for consisency
+        method = method.upper()
+
+        if method not in VALID_METHODS:
+            raise Exception(f'Invalid method {method}')
+
+        # GET/PUT requests have the data interpolated into the url
+        if method in ['GET', 'PUT']:
+            endpoint = endpoint.format(data)
+
+        url = f"{self.base_url}/{endpoint}"
+        all_headers = {**self.headers, **headers} if headers else self.headers
+        response = requests.request(method, url, data=data, headers=all_headers)
+
+        if response.status_code == 200:
+
+            # Return JSON if that's what it is (this should be the default)
+            if response.headers.get('Content-Type', str) == 'application/json':
+                return response.json()
+            
+            return response
+
+        else:
+            raise Exception(f"Request failed with status code {response.status_code}: {response.text}")
+
+
+    def login(self, email: str, password: str):
+        """Log the user into ME.org.
+
+        Parameters
+        ----------
+        email : str
+            Registered email address.
+        password : str
+            Password (will be hashed)
+
+        Raises
+        ------
+        Exception
+            When the login fails. 
+        """
+
+        # Assemble payload
+        login_data = {
+            'email': email,
+            'password': hl.sha256(password.encode('UTF-8')).hexdigest(),
+            'hashed': 'true'
+        }
+
+        # Call
+        response = self._make_request('post', endpoint='login', data=login_data)
+
+        # Successful login
+        if response.get('status', 'error') == 'success':
+            auth_headers = {
+                'X-User-Id': response['data']['userId'],
+                'X-Auth-Token': response['data']['authToken']
+            }
+
+            self.headers.update(auth_headers)
+
+        # Unsuccessful login
+        else:
+
+            raise Exception("Login failed")
+
+
+    def get_file_status(self, file_id: str) -> dict:
```

Comment:
We can use the [`typing.Union`](https://docs.python.org/3/library/typing.html#typing.Union) type here:
```suggestion
    def get_file_status(self, file_id: str) -> Union[dict, requests.Response]:
```